From: Jens Axboe <jens.axboe@oracle.com>
Date: Tue, 3 Nov 2009 10:40:44 +0000 (+0100)
Subject: fuse: fix kunmap in fuse_ioctl_copy_user
X-Git-Tag: v2.6.32-rc7~67^2~1
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=0bd87182d3ab18a32a8e9175d3f68754c58e3432

fuse: fix kunmap in fuse_ioctl_copy_user

Looks like another victim of the confusing kmap() vs kmap_atomic() API
differences.

Reported-by: Todor Gyumyushev <yodor1@gmail.com>
Signed-off-by: Jens Axboe <jens.axboe@oracle.com>
Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
Cc: Tejun Heo <tj@kernel.org>
Cc: stable@kernel.org
---

diff --git a/fs/fuse/file.c b/fs/fuse/file.c
index 5887a63..c18913a 100644
--- a/fs/fuse/file.c
+++ b/fs/fuse/file.c
@@ -1600,7 +1600,7 @@ static int fuse_ioctl_copy_user(struct page **pages, struct iovec *iov,
 			kaddr += copy;
 		}
 
-		kunmap(map);
+		kunmap(page);
 	}
 
 	return 0;
